### PR TITLE
Addresses issues from 0.21.6

### PR DESCRIPTION
**Pull request by sbrossie**

### CODE REVIEW

File: invoice/src/main/java/org/killbill/billing/invoice/usage/RawUsageOptimizer.java [hunk 1]
```diff
@@ -122,7 +122,9 @@ public BillingPeriod apply(final Usage input) {
             final InvoiceItem item = iterator.previous();
             if (!(item instanceof UsageInvoiceItem)) {
                 // Help to debug https://github.com/killbill/killbill/issues/1095
-                log.warn("RawUsageOptimizer : item id={}, expected to see an UsageInvoiceItem type, got {}", item.getId(), item);
+                log.warn("RawUsageOptimizer : item id={}, expected to see an UsageInvoiceItem type, got class {}",
+                         item.getId(),
+                         (item != null ? item.getClass() : null));
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/usage/RawUsageOptimizer.java]: Maybe you should also print the class loader?

File: invoice/src/main/java/org/killbill/billing/invoice/generator/InvoicePruner.java [hunk 2]
```diff
@@ -185,22 +185,36 @@ private void build() {
...
+            } else {
+                final BigDecimal remainingFromRepair = target.getAmount().subtract(repairedAmount);
+                final int compRepair = remainingFromRepair.compareTo(BigDecimal.ZERO);
+
+
+                // We repaired the whole thing
+                if (compRepair == 0) {
+                    // Keep previous precondition behavior and check there is no extra adjustment on top of it.
+                    // @see TestIntegrationInvoiceWithRepairLogic#testWithFullRepairAndExistingPartialAdjustment
+                    Preconditions.checkState(adjusted == null || adjusted.isEmpty(), "Too many repairs for invoiceItemId='%s'", target.getId());
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/generator/InvoicePruner.java]: Maybe print the `adjusted` in the error message

File: invoice/src/main/java/org/killbill/billing/invoice/usage/RawUsageOptimizer.java (hunk 1 above)
Comment (sbrossie) [reply invoice/src/main/java/org/killbill/billing/invoice/usage/RawUsageOptimizer.java]: See https://github.com/killbill/killbill/pull/1247/commits/0139d1462046a1fbf4d3bc60fb0557cf21cdac51

File: invoice/src/main/java/org/killbill/billing/invoice/generator/InvoicePruner.java (hunk 2 above)
Comment (sbrossie) [reply invoice/src/main/java/org/killbill/billing/invoice/generator/InvoicePruner.java]: See https://github.com/killbill/killbill/pull/1247/commits/0139d1462046a1fbf4d3bc60fb0557cf21cdac51

**Pull request merged**